libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -407,3 +409,24 @@ void ConstantEvaluator::endVisit(TupleExpression const& _tuple)
 	if (!_tuple.isInlineArray() && _tuple.components().size() == 1)
 		m_values[&_tuple] = evaluate(*_tuple.components().front());
 }
+
+void ConstantEvaluator::endVisit(MemberAccess const& _memberAcess)
+{
+	if (auto const* parentIdentifier = dynamic_cast<Identifier const*>(&_memberAcess.expression()))
+	{
+		if (auto const* contract = dynamic_cast<ContractDefinition const*>(parentIdentifier->annotation().referencedDeclaration))
+		{
+			auto contractVariables = contract->stateVariables();
+			auto variable = ranges::find_if(
+				contractVariables,
+				[&](VariableDeclaration const* _variable) { return _variable->name() == _memberAcess.memberName(); }
```

Comment:
Can there be a problem with shadowing?

Reply:
I don't think there can be. The contract name cannot be shadowed and then the variable name is searched among the state variables of the contract only.
I tried to declare something to shadow the contract name, such as a struct, error, another contract in another file, but the compiler would not accept using the same name.

Reply:
Thanks for investigating!